### ISSUE TITLE
Prior art in this area

### ISSUE THREAD

**Issue by jordansissel** [evanphx/eventd-rfc#4]
With respect to protocols/formats, there are two efforts I know of (but there are probably more):
- MITRE CEE (https://cee.mitre.org/language/1.0-beta1/)
- ArcSight CEF

Both of these projects (CEE and CEF), in my opinion, case studies as things that failed to really solve any problems. I only put them forward here as lessons to learn with respect to log formats and protocols.

**Comment by evanphx** [comment]
CEE is certainly interesting and I'm really sad to see it basically die. I think it's a very straightforward approach to applying a JSON schema to logged event data.

I'll check out CEF today.

I think the broader discussion we should have is why did these do studies fail? Was it because they were designed in a vacuum, without participation from the people that matter in the space (the implementors of logging systems)?

**Comment by jordansissel** [comment]
@evanphx my reading of CEE was much more dystopian, hah. I summarized my feelings [here](http://www.semicomplete.com/blog/geekery/CEE-logging-for-profit.html) a few years ago, but the gist is that there were too many serializations (xml, json, etc), too many protocols (tcp, and more), and very inconsistent naming. I loved the idea of CEE (common grounds log transport and format), but the specification had too much incompatibility opportunities.

**Comment by jordansissel** [comment]
@evanphx My choice of calling CEE a failure was an inarticulate summary of my own opinion on CEE. The project itself is dead (defunded?) at MITRE, but some of the ideas were carried forward, I think, by the rsyslog folks under the project name 'lumberjack' - https://fedorahosted.org/lumberjack/ - though I haven't followed that project in years.

**Comment by evanphx** [comment]
@jordansissel I've forgotten about xml in CEE, ha! Perhaps that was on purpose. I agree that there can't be too many serialization formats and none of them can be xml at this stage.

Yes, lumberjack also appears dead.

So the thing to learn from CEE is that the specification needs to be small, even at the expensive of flexibility. Otherwise it's too big for people to implement.

And from lumberjack that community discussion in the open is key.

**Comment by jordansissel** [comment]
Agreed there - my main learning from CEE is that there are 3 major areas to focus on - serialization, naming, and transit protocol.

**Comment by josegonzalez** [comment]
Also related is the metrics 2.0 spec: http://metrics20.org/

**Comment by evanphx** [comment]
@josegonzalez I've added metrics2.0.

**Comment by sandfox** [comment]
Is it worth adding anything from/about heroku's logplex, log-shuttle, and log-iss even if they are mostly just dealing with the problems of transport rather than formatting (still basically syslog format)

https://github.com/heroku/logplex/tree/master/doc

**Comment by evanphx** [comment]
@sandfox It definitely is! It's a big omission on my part. I'm heavily inspired by logplex but somehow neglected to add it! That will be fixed today.
